test/acceptance/application.test.ts

```diff
@@ -19,375 +20,82 @@ describe('Controllers', () => {
-                  title: 'jump the gun'
-                }
-              ]
-            })
-          )
-          .expect(400);
-      });
-      it('throws if the payload is missing a title', async () => {
-        await client
-          .post('/todo')
-          .send({})
-          .expect(400);
-      });
-
-      it('returns 409 if the item already exists', async () => {
-        const todo = (await client.post('/todo').send(payload)).body;
-        await client
-          .post('/todo')
-          .send(todo)
-          .expect(409);
-      });
-    });
-
-    describe('findById', () => {
-      it('returns a todo if it exists', async () => {
-        const item = await client.post('/todo').send({
-          title: 'fake'
-        });
-        await client
-          .get(`/todo/${item.body.id}`)
-          .send()
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .get(`/todo/9999999`)
-          .send()
-          .expect(404);
-      });
-    });
-
-    describe('find', () => {
-      it('finds everything', async () => {
-        await client
-          .get('/todo')
-          .send()
-          .expect(200);
-      });
-    });
-
-    describe('replace', () => {
-      it('successfully replaces existing items', async () => {
-        const item = await client.post('/todo').send({
-          title: 'another one'
-        });
-
-        await client
-          .put(`/todo/${item.body.id}`)
-          .send({
-            title: 'another two'
-          })
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .put('/todo/999999')
-          .send({
-            title: 'another one'
-          })
-          .expect(404);
-      });
-    });
-
-    describe('update', () => {
-      it('successfully updates existing items', async () => {
-        const item = await client.post('/todo').send({
-          title: 'another one'
-        });
-
-        await client
-          .patch(`/todo/${item.body.id}`)
-          .send({
-            title: 'another two'
-          })
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .patch('/todo/999999')
-          .send({
-            title: 'another one'
-          })
-          .expect(404);
-      });
-    });
-
-    describe('delete', () => {
-      it('successfully deletes matching item', async () => {
-        await client
-          .del('/todo/1')
-          .send()
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .del('/todo/999999')
-          .send({
-            title: 'another one'
-          })
-          .expect(404);
-      });
-    });
+  const payload = {
+    title: 'do a thing',
+    desc: 'There are some things that need doing',
+    isComplete: false
+  };
```

Comment:
Have you considered using data-builder here? See http://loopback.io/doc/en/lb4/Testing-your-application.html#use-test-data-builders

Feel free to make this change later, after this long-living PR is landed.

Reply:
I think I'll make that change as a part of this one, I agree that it's a nicer pattern.